### ISSUE TITLE
VHDL: procedures inside processes (Origin: bugzilla #777648)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#6151]
status NEW severity *normal* in component *build* for *---*
Reported in version *1.8.12* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[bug_project.zip](http://ftp.stack.nl/pub/users/dimitri/bugzilla/attachments//bug777648_344036_bug_project.zip)_ (ID 344036)
- _[vhdl.patch](http://ftp.stack.nl/pub/users/dimitri/bugzilla/attachments//bug777648_344991_vhdl.patch)_ (ID 344991)

On 2017-01-23 14:37:42 +0000, Jorge T wrote:

> Created attachment 344036
> vhdl example + doxygen config file
> 
> Hi!
> 
> When using procedures inside processes (for VHDL testbench development), the HTML output is not generated correctly. 
> 
> If the procedures are commented, doxygen works fine.

On 2017-02-05 18:34:12 +0000, mk wrote:

> Created attachment 344991
> patch for this bug
> 
> this patch should fix the bug

**Comment by jtonfat** [comment]
I would like to remark that the following bug is still happening with Doxygen 1.8.17.

The problem is related to the usage of procedures inside processes. This is a common usage for testbench development.
The problem is that the \brief and \details information are not assigned to the correct element. The documentation of the vhdl process is assigned to the first procedure.
I am attaching an example +doxygen config file that reproduce this behavior.

[bug_project2020.zip](https://github.com/doxygen/doxygen/files/4193299/bug_project2020.zip)

**Comment by albert-github** [comment]
I can reproduce the problem, also with the current master (d0705323ef3fdc1db315ee8cd8326a7d0da5ced7).


One remark, on line 48 you wrote:
```
\arg \ref a - ..
```
I think this should be:
```
\param a ...
```

**Comment by albert-github** [comment]
I did some small tests and it looks like doxygen gets confused by the comment on the lines 35 / 36 just above `genInputProc: process is`. When I make instead of `--!` a `--` comment out of it the comment is placed at the right place, but of course we loose the comment from the lines 35 / 36.

**Comment by jtonfat** [comment]
> I can reproduce the problem, also with the current master ([d070532](https://github.com/doxygen/doxygen/commit/d0705323ef3fdc1db315ee8cd8326a7d0da5ced7)).
> 
> One remark, on line 48 you wrote:
> 
> ```
> \arg \ref a - ..
> ```
> 
> I think this should be:
> 
> ```
> \param a ...
> ```

Yes, you are right. In this case should be \param, but still the problem is there.
